internal/pkg/agent/application/upgrade/upgrade.go

```diff
@@ -328,6 +324,26 @@ func (u *Upgrader) Upgrade(ctx context.Context, version string, sourceURI string
 	return cb, nil
 }
 
+func getParsedVersionWithFallback(log *logger.Logger, currentVersion string, fallbackVersion *agtversion.ParsedSemVer) *agtversion.ParsedSemVer {
+	previousParsedVersion, err := agtversion.ParseVersion(currentVersion)
+	if err != nil {
+		previousParsedVersion = fallbackVersion
+		log.Warnw(fmt.Sprintf("parsing of current version failed, watcher selection will fall back to comparing to %q", previousParsedVersion), "error.message", err)
+	}
+	return previousParsedVersion
+}
+
+func selectWatcherExecutable(topDir string, previous agentInstall, current agentInstall) string {
+	// check if the upgraded version is less than the previous (currently installed) version
+	if current.parsedVersion.Less(*previous.parsedVersion) {
+		// use the current agent executable for watch, if downgrading the old agent doesn't understand the current agent's path structure.
```

Comment:
comment says use the current as currently installed but it's not current as a variable name. it gets confusing

Reply:
Agreed but this is the existing names that follow where the symlink points.
So after rotating the symlink the currently running agent is named previous and the new one is named current.
I don't know if this is more confusing or swapping the variable names so when we look at it in different places in the code it gets confusing...

Reply:
can we maybe rephrase
`use the current agent executable for watch, if downgrading the old agent doesn't understand the current agent's path structure.`
to 
`if downgrading the target agent doesn't understand the current agent's path structure.`